### PR TITLE
add subscribe method

### PR DESCRIPTION
**Pull request by vorobeez**

### CODE REVIEW

File: packages/api/src/Api.ts [hunk 1]
```diff
@@ -68,13 +74,17 @@ export class Api {
     this.catchErrors = catchErrors;
     this.requestHandler = ChainHandler.fromArray(transformRequest);
     this.responseHandler = ChainHandler.fromArray(transformResponse);
+    this.subscriptionClient = this.prepareSubscriptionClient(
```
Comment (vorobeez) [inline packages/api/src/Api.ts:77]: @sebscholl Here's the thing.

We have two entities for graphql subscriptions.
1. A subscription client which is a global connection to a server for all subscriptions. And it's got `connecting/connected/reconnecting/reconnected/disconnected/error` callbacks. See `prepareSubscriptionClient` method for initialization details. You can close connection with `closeSubscriptionConnection` method
2. A subscription request. It's just only one subscription query which can return some data. It's got `data/error` callbacks only. And it returns `unsubscribe` function.

Comment (sebscholl) [reply packages/api/src/Api.ts:77]: So can you help me understand then how the two work together? 

For example, as a SDK user must I initialize the first subscription client, specifying all my callbacks (that I want), and then I am able to use the request method and specify a data/error callback?

Comment (vorobeez) [reply packages/api/src/Api.ts:77]: That's right. Here's an example.

```
api = new Api({
  workspaceId: 'workspaceId',
  subscription: {
    connected: () => {
      console.log('connection has estalished!');
    },
  },
});

api.subscription(
  `
  subscription {
    ApiTests {
      node {
        id
        timestamp
      }
    }
  }
  `,
  { 
    data: (result) => { console.log(result), }
  }
);
```

Comment (vorobeez) [reply packages/api/src/Api.ts:77]: @sebscholl Can I merge it? Or maybe you have questions or comments?

Comment (sebscholl) [reply packages/api/src/Api.ts:77]: Merge it!

**Pull request merged**